guides/source/routing.md

```diff
@@ -18,51 +18,66 @@ After reading this guide, you will know:
 The Purpose of the Rails Router
 -------------------------------
 
-The Rails router recognizes URLs and dispatches them to a controller's action, or to a Rack application. It can also generate paths and URLs, avoiding the need to hardcode strings in your views.
+Rails routing is a very powerful and flexible feature. The Rails router
+dispatches incoming HTTP requests to specific controller actions in your Rails
+application based on the URL path. (It can also dispatch to a
+[Rack](rails_on_rack.html) application.) It also generates path and URL helpers
+that can be used in your application (instead of hard-coding strings) based on
+the resources configured in the router.
 
-### Connecting URLs to Code
+### Routing Incoming URLs to Code
 
 When your Rails application receives an incoming request for:
 
 ```
-GET /patients/17
+GET /users/17
 ```
 
 it asks the router to match it to a controller action. If the first matching route is:
 
 ```ruby
-get '/patients/:id', to: 'patients#show'
+get '/users/:id', to: 'user#show'
 ```
 
-the request is dispatched to the `patients` controller's `show` action with `{ id: '17' }` in `params`.
+the request is dispatched to the `users` controller's `show` action with `{ id: '17' }` in `params`.
 
-NOTE: Rails uses snake_case for controller names here, if you have a multiple word controller like `MonsterTrucksController`, you want to use `monster_trucks#show` for example.
+Passing a `String` to `to:` will expect a `controller#action` format. When using a `Symbol`, the `to:` option should be replaced with `action:`. When using a `String` without a `#`, the `to:` option should be replaced with `controller:`:
+
+```ruby
+get '/users/:id', action: :show, controller: 'users'
+```
+
+NOTE: Rails uses snake_case for controller names, if you have a multiple word controller like `UserProfilesController`, you'd use `user_profiles#show` when specifying the route.
 
 ### Generating Paths and URLs from Code
```

Comment:
This section confuses me a little bit because initially, I thought we were explaining how the automatically generated paths and URLs work in Rails routing but then in the example we use "as:" to specify a name for the route because we're using a "/get". Maybe there's some way to add clarity on that.

Reply:
Yes added a note to clarify that